### PR TITLE
yes(): abort on wrong answers, saying so

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
except for the passphrase display as we can only display it as long as we have it in memory,
here: retry, telling the user if he entered something invalid and needs to enter again.

### CODE REVIEW

File: borg/archiver.py [hunk 1]
```diff
@@ -466,8 +467,8 @@ def do_delete(self, args, repository):
                         msg.append(format_archive(archive_info))
                 msg.append("Type 'YES' if you understand this and want to continue: ")
                 msg = '\n'.join(msg)
-                if not yes(msg, false_msg="Aborting.", truish=('YES', ),
-                           env_var_override='BORG_DELETE_I_KNOW_WHAT_I_AM_DOING'):
+                if not yes(msg, false_msg="Aborting.", invalid_msg='Invalid answer, aborting.', truish=('YES', ),
```
Comment (RonnyPfannschmidt) [inline borg/archiver.py]: whops, i meant why is there no default prompt?

Comment (ThomasWaldmann) [inline borg/archiver.py]: yes() does not have any builtin messages. it also does not default to emitting messages at the various places. if you want to see some msg, you have to give it. also, caller is responsible for i18n, if wanted.

Comment (RonnyPfannschmidt) [inline borg/archiver.py]: if callers must provide, then why supply empty default arguments where programmer mistakes lead to ux fail?

Comment (ThomasWaldmann) [inline borg/archiver.py]: maybe we could use non-language defaults.
like "> " and "???" for invalid or so.
i don't want yes() to get into i18n department.

**Comment by ThomasWaldmann** [comment]
merge?

**Comment by ThomasWaldmann** [comment]
yup.

**Pull request merged**